src/backend/tests/unmocked/onboarding.test.ts

```diff
@@ -75,6 +76,108 @@ describe('Onboarding tests', () => {
     });
   });
 
+  describe("Get User's Checklists", () => {
+    it('Returns all general checklists when checklist fails to match user teamType', async () => {
+      createTestTeamType('id', organization);
+      const user = await createTestUser(batmanAppAdmin, orgId);
+
+      // Create a team and add the user as a member (getUsersChecklists requires user to be on a team)
+      const team = await TeamsService.createTeam(
+        user,
+        'Engineering Team',
+        user.userId,
+        'slackId123',
+        'This is the engineering team',
+        false,
+        organization
+      );
+      await prisma.team.update({
+        where: { teamId: team.teamId },
+        data: { members: { connect: { userId: user.userId } } }
+      });
+
+      // Create a checklist, set its teamTypeId to null to be a general checklist
+      const checklist = await createTestChecklist(user, orgId);
+      const generalChecklist = await prisma.checklist.update({
+        where: { checklistId: checklist.checklistId },
+        data: { teamTypeId: null },
+        include: { checklistItems: true }
+      });
+
+      const result = await OnboardingServices.getUsersChecklists(user.userId);
+      expect(result).toStrictEqual([generalChecklist]);
+    });
+
+    it('Returns all general checklist and matching checklists of user teamType', async () => {
+      const teamType = createTestTeamType('id', organization);
+      const user = await createTestUser(batmanAppAdmin, orgId);
+
+      // Create a team and add the user as a member
+      const team = await TeamsService.createTeam(
+        user,
+        'Engineering Team',
+        user.userId,
+        'slackId123',
+        'This is the engineering team',
+        false,
+        organization
+      );
+      await prisma.team.update({
+        where: { teamId: team.teamId },
+        data: {
+          members: { connect: { userId: user.userId } },
+          teamType: { connect: { teamTypeId: (await teamType).teamTypeId } }
+        }
+      });
+
+      // Create a checklist, set its teamTypeId to null to be a general checklist
+      const checklist = await createTestChecklist(user, orgId);
+      const generalChecklist = await prisma.checklist.update({
+        where: { checklistId: checklist.checklistId },
+        data: { teamTypeId: null },
+        include: { checklistItems: true }
+      });
+
+      // Create a checklist that matches the user's team type
+      const teamChecklist = await createTestChecklist(user, orgId);
+
+      const result = await OnboardingServices.getUsersChecklists(user.userId);
+      expect(result).toEqual([generalChecklist, teamChecklist]);
+    });
+
+    it('Throws an error if the user does not have any teams', async () => {
```

Comment:
Move this to be the first test, usually we have all of our "failing" tests first and then the successful ones